Added site creation page

gsidebo opened the pull request:
#### Pre-Flight checklist

- [x] Screenshots and design review for any changes that affect layout or styling
- [x] Testing
  - [x] Code is tested
  - [x] Changes have been manually tested

#### What are the relevant tickets?
Closes #45 

#### What's this PR do?
- Adds the site creation page
- Adds the formik library

#### How should this be manually tested?
- Make sure you have at least 1 starter in your db
- Navigate to `/sites/`
- Enter a title and submit

A new website should be created, and you should be redirected to the detail page for that site right away.

#### Any background context you want to provide?
- I removed pagination on the WebsiteStarter API results. We can add it back if we need it, but it seems unlikely that we'd have so many starters that we would really need to page through them
- I changed up the styling of the sidebar + content in the site page. There were some quirks in the way the elements were lining up with the header, and while fixing that I realized that it doesn't make much sense for the sidebar to grow proportionally with the browser window, so I changed it to use flexbox instead of a bootstrap container+columns
- I'm hoping we can establish some patterns for (a) testing forms, and (b) displaying API response errors as form errors. The patterns used in this PR should get some extra scrutiny so we can be sure we're comfortable with them for future use cases.

#### Screenshots (if appropriate)
![ss 2021-02-24 at 15 27 59 ](https://user-images.githubusercontent.com/14932219/109064488-77bf0700-76b8-11eb-97ce-271ad13189b8.png)
![ss 2021-02-24 at 15 49 57 ](https://user-images.githubusercontent.com/14932219/109064493-7ab9f780-76b8-11eb-859b-2c7c278494f4.png)

mbertrand commented on static/js/pages/SitesDashboard.tsx:
```diff
@@ -0,0 +1,84 @@
+      const errors = getResponseBodyErrors(response)
+      if (!errors) {
+        return
+      } else if (isString(errors)) {
```
FWIW, I got some feedback on several PRs awhile back that we should try to avoid lodash and use ramda if at all possible.  `R.is(String, errors)` should probably work.  Though personally I find lodash much easier to understand 😁

gsidebo replied:
If nothing else it will be good to cut down on the number of libraries

noisecapella reviewed (changes requested):
Functionality works great, just a few comments

mbertrand commented on static/js/components/forms/SiteForm.test.tsx:
```diff
@@ -0,0 +1,73 @@
+    it("rejects an empty title", async () => {
+      try {
+        await expect(
+          await websiteValidation.validateAt("title", "")
```
Although this test passes, I don't think it's working as intended.  If you change it to `websiteValidation.validateAt("title", "Some Title")` it will still throw a "required field" error.  Ran into this while testing my PR using the same approach.

mbertrand replied:
```python
validateAt("title", {title: ""})
```

gsidebo commented:
@noisecapella I fixed those issues, and also added a migration to make `Website.title` required. Didn't make much sense to ask for the title and autogenerate the name/slug from it if it wasn't required in the db

gsidebo commented:
@noisecapella Ready for review again
